base/ApplicationSupport.F90

```diff
@@ -56,8 +55,7 @@ subroutine MAPL_Finalize(unusable,comm,rc)
       _UNUSED_DUMMY(unusable)
       
       if (present(comm)) then
-         call MPI_comm_dup(comm,comm_world,status)
-         _VERIFY(status)
+         comm_world = comm
```

Comment:
Why does Finalize need to dup a communicator at all?

Reply:
I believe it is a bug to duplicate here

Reply:
Even we don't duplicate here, it is still a bug because the original comm was duplicated. Remove those mpi_comm_dups may be the best bet.